src/Docker.DotNet/Models/PluginConfigInterface.Generated.cs

```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace Docker.DotNet.Models
+{
+    [DataContract]
+    public class PluginConfigInterface // (types.PluginConfigInterface)
+    {
+        [DataMember(Name = "Socket", EmitDefaultValue = false)]
+        public string Socket { get; set; }
+
+        [DataMember(Name = "Types", EmitDefaultValue = false)]
+        public IList<PluginInterfaceType> Types { get; set; }
```

Comment:
This doesn't work for log drivers. The "Types" return for log driver is a string instead of PluginInterfaceType object.
Any ideas how to make sure this works for log drivers as well?

Reply:
Are you sure? Based on the spec I don't see how the go code would handle that case either. This is the correct generated code for the object type `PluginInterfaceType` which would be deserialized from json `object` not string.

Reply:
Yes, I am sure,
I am implementing the API for supporting my custom log driver plugin. When I tried to list all the plugins, the code failed saying we cannot convert string to PluginInterfaceTypes object.
On using IList<string> the API worked fine.

Reply:
Can you paste what the JSON array is thats being returned in raw text? I would love to see what the daemon is sending.

Reply:
[{"Config":{"Args":{"Description":"","Name":"","Settable":null,"Value":null},"Description":"Send log messages to elasticsearch","DockerVersion":"17.05.0-ce","Documentation":"-","Entrypoint":["/usr/bin/docker-log-elasticsearch"],"Env":[{"Description":"Set log level to output for plugin logs","Name":"LOG_LEVEL","Settable":["value"],"Value":"info"}],"Interface":{"Socket":"elasticsearchlog.sock","Types":["docker.logdriver/1.0"]},"IpcHost":false,"Linux":{"AllowAllDevices":false,"Capabilities":null,"Devices":null},"Mounts":null,"Network":{"Type":"host"},"PidHost":false,"PropagatedMount":"","User":{},"WorkDir":"","rootfs":{"diff_ids":["sha256:bd7b32cc16b423b23e33f6c770b5ef044f78af8b250f79d8e9c818bc07745394"],"type":"layers"}},"Enabled":true,"Id":"5216a226888623f939799f4ee640c88a399a621906a2c7f38e255170cabd738e","Name":"elasticsearch:latest","PluginReference":"docker.io/rchicoli/docker-log-elasticsearch:latest","Settings":{"Args":[],"Devices":[],"Env":["LOG_LEVEL=info"],"Mounts":[]}}]

Reply:
Here is the culprit: [plugin_response.go](https://github.com/moby/moby/blob/3a633a712c8bbb863fe7e57ec132dd87a9c4eff7/api/types/plugin_responses.go#L44). They have cheated :)! You have two options. Either change specgen.go to customize the "Types" field to be a string (easiest option) or you will need to build the JSON.Net Serialize/Deserialize code for this C# generated type that does the same .ToString()/Parse logic. Make sense?

Reply:
How do I make "Types" field a string in specgen.go?

Reply:
Thanks. Will make the change soon.